gitattributes: auto resolve merge conflicts for generated files

sekhar-isovalent opened the pull request:
Let git automatically choose a dummy merge strategy for our generated files to streamline the process of release prep.

sekhar-isovalent committed:
.gitattributes: auto resolve merge conflicts for generated files

Signed-off-by: Sekhar Sankaramanchi <sekhar@isovalent.com>

bimmlerd commented on .gitattributes:
```diff
@@ -16,3 +16,7 @@ pkg/k8s/client/clientset/** linguist-generated
 pkg/k8s/client/listers/** linguist-generated
 *.bt linguist-language=D
+
+./install/kubernetes/cilium/Chart.yaml merge=ours
```
I don't think `Chart.yaml` is autogenerated, and hence I don't think a `merge=ours` is a good idea.

sekhar-isovalent replied:
Taken directly from `.github/ISSUE_TEMPLATE/enterprise_release_template.md?plain=1#L55`. May be we should update it there as well ?

bimmlerd replied:
Well, no - for a release manager, the changes made to the chart are autogenerated and it's fine to take 'ours' to resolve the final conflict. But for engineering in general the merge behaviour shouldn't be that.

sekhar-isovalent replied:
Updated PR

sekhar-isovalent committed:
remove `install/kubernetes/cilium/Chart.yaml` from `.gitattributes`

Since `Chart.yaml` is not auto-generated, remove it from merge strategy handling.

Signed-off-by: Sekhar Sankaramanchi <sekhar@isovalent.com>

joestringer commented on .gitattributes:
```diff
@@ -16,3 +16,6 @@ pkg/k8s/client/clientset/** linguist-generated
 pkg/k8s/client/listers/** linguist-generated
 *.bt linguist-language=D
+
+./install/kubernetes/cilium/values.yaml merge=ours
```
Should this line be combined with the line at the top or should it be separate?

Is the `./` prefix vs. `/` prefix vs. no `./` meaningful?
